src/testdrive/src/error.rs

```diff
@@ -57,13 +57,16 @@ pub enum Error {
 }
 
 impl Error {
-    pub fn print_stderr(&self) -> io::Result<()> {
+    pub fn print_stderr(&self, ci_output: bool) -> io::Result<()> {
```

Comment:
Minor, but seems a little weird to pass this through to this method since I don't think in general you want the ability to print an error followed by a Buildkite log output thingy?

Reply:
Sort of? The only place that this function is called is in `main()` and it's exclusively used for handling messaging "we have failed the tests", which is where you want this.

Yeah the three options I thought of were:

* put the buildkite output in `main()` -- felt wrong because then you've got two completely separate parts of the code responsible for  the same display
* modifying Error to take a `ci_output` as a field -- would have required a pretty large rewrite, and error shouldn't really know about its output format
* this change.

Open to other ideas, though.

Reply:
The first bullet is what I was thinking! IMO then you have an error object that's only responsible for printing itself to the screen, and a main function that's doing the separate task of telling Buildkite about the error. But this is fine!